src/main/io/osd.c

```diff
@@ -668,6 +671,21 @@ static bool osdDrawSingleElement(uint8_t item)
                 break;
             }
 
+#ifdef USE_ESC_SENSOR
+            // Show warning if we lose motor output
+            if (enabledWarnings & OSD_WARNING_ESC_FAIL && ARMING_FLAG(ARMED)) {
+                for (int i = 0; i < getMotorCount(); i = i + 1) {
+                    escSensorData_t *escDatai = getEscSensorData(i);
+                    if (escDatai->rpm == 0) {
+                        char escErrMsg[OSD_FORMAT_MESSAGE_BUFFER_SIZE] = "";
+                        tfp_sprintf(escErrMsg, "ESC %2d ERR", i + 1);
```

Comment:
Anyone flying with 10+ motors? ;)
(`ESC 124 ERR` or `ESC 12 4 ERR` may be beter)

Reply:
We used to have `MAX_SUPPORTED_MOTORS` set to 12, that's probably where the legacy code supporting > 9 motors comes from.

There are still some development board targets supporting more than 9 motors, but we can probably limit those to 9, so that we can start changing code to expect single digit motor numbers.

Reply:
@pulquero: I have given this some thoughts:

I think @ledvinap's suggestion to show **all** failed ESCs instead of just the first one is good - getting the complete picture is valuable.

Since almost all targets only support 8 or even 4 motors, you can generate the list of failed ESCs conditional on `MAX_SUPPORTED_MOTORS <= 8`, and use an empty string for the exotic targets that go over this.

Taking into account that the maximum string size for warnings is 11, the resulting string would have to look like `ESC1 34`. Not pretty, but useful.

Reply:
Ahh, ok, now I understand. I'll do the positional ESC1 34 style then.

Reply:
Sorry for not making this clear from the start. As an example, on an octocopter, `ESC 2    7 ` would mean 'problem with ESCs 2 and 7 detected.